packages/redux-store/src/recordListener.ts

```diff
@@ -21,7 +26,21 @@ export const removeRecord = createAction<BaseRecord>('record/remove')
  */
 export const startRecordListeners = (agent: Agent, store: Store) => {
   const onDeleted = (event: RecordDeletedEvent<BaseRecord>) => {
-    store.dispatch(removeRecord(event.payload.record))
+    // Extract the record
+    const record = event.payload.record
+    // Retrieve and delete the record if it is of the generic type
+    const retrieveAndDeleteGeneric = async (record: GenRecord) => {
```

Comment:
@blu3beri not quite sure here (see also [your old comment](https://github.com/hyperledger/aries-framework-javascript-ext/pull/204/files#r1153021830). Might be butchering this. Let me know if you have other things in mind.

Reply:
`genericRecords` would not be able to fetch credentialexchangerecords, proofexchangerecords, etc.

The record might be a type that extends the `BaseRecord` or its `{ id:string, type string }`. 

If it extends a `BaseRecord`, the previous flow was fine. If its an `{id: string, type: string}` it must be fetched manually.

To fetch it maually, although this is getting into some odd dependency injection stuff here so I might get some details wrong, you can resolve the `storageService` from the agent and call `getById`,
https://github.com/hyperledger/aries-framework-javascript/blob/644e860a05f40166e26c497a2e8619c9a38df11d/packages/core/src/storage/StorageService.ts#L62, with the Record Class based on the type. 

@TimoGlastra @genaris I remember something vaguely that this was done to avoid another`getById` call for the `deleteById` method, but if this is the correct way of doing it, it seems like quite a clunky API.

Reply:
There's not really a need to remove the record here. This event is emitted when the record is already deleted, and the only thing we need to do is to remove it from the redux store.

I think we should just match the API from AFJ and allow either a record, or an object with `type` and `id` to be passed, as that's enough to remove something from the store.

I think we can update the `removeRecord` method to something like this. Haven't tested it, but should work. If it's really difficult. the removeRecord could also just take an id instead of the whole record, but I've made it a record to keep it consistent with the add / update methods.

```
export const removeRecord = <R extends BaseRecordAny>(record: R | { id: string; type: R['type'] }, state: RecordsState<R>): RecordsState<R> => {
```